modules/hal_nordic/nrf_802154/sl_opensource/platform/nrf_802154_lp_timer_zephyr.c

```diff
@@ -99,99 +74,83 @@ void nrf_802154_lp_timer_deinit(void)
-
-	m_in_critical_section = false;
-
-	z_nrf_rtc_timer_compare_int_unlock(m_rtc_channel, m_is_running);
-
-	nrf_802154_sl_mcu_critical_exit(state);
+	if (round_up) {
+		return NRF_802154_SL_US_TO_RTC_TICKS(us);
+	} else {
+		return NRF_802154_SL_US_TO_RTC_TICKS(us) - 1ULL;
+	}
 }
 
-uint32_t nrf_802154_lp_timer_time_get(void)
+uint64_t nrf_802154_platform_sl_lptimer_lpticks_to_us_convert(uint64_t lpticks)
 {
-	/* Please note that this function does not handle RTC overflow */
-	return NRF_802154_SL_RTC_TICKS_TO_US(z_nrf_rtc_timer_read());
+	/* Calculations are performed on uint64_t as it is safe to assume
+	 * overflow will not occur in any foreseeable future.
+	 */
+	return NRF_802154_SL_RTC_TICKS_TO_US(lpticks);
 }
 
-uint32_t nrf_802154_lp_timer_granularity_get(void)
+void nrf_802154_platform_sl_lptimer_schedule_at(uint64_t fire_lpticks)
 {
-	return NRF_802154_SL_US_PER_TICK;
-}
+	/* This function is not required to be reentrant, hence no critical section. */
+	if ((m_last_scheduled_time != fire_lpticks) || !m_is_running) {
+		m_last_scheduled_time = fire_lpticks;
+		m_is_running = true;
+		__DMB();
 
-void nrf_802154_lp_timer_start(uint32_t t0, uint32_t dt)
-{
-	timer_start_at(m_rtc_channel, t0, dt);
+		z_nrf_rtc_timer_set(m_rtc_channel, fire_lpticks, rtc_irq_handler, NULL);
+	}
 }
 
-bool nrf_802154_lp_timer_is_running(void)
+void nrf_802154_platform_sl_lptimer_disable(void)
 {
-	return m_is_running;
+	/* This function is not required to be reentrant, hence no critical section. */
+	(void)z_nrf_rtc_timer_compare_int_lock(m_rtc_channel);
+
+	m_is_running = false;
+	__DMB();
 }
 
-void nrf_802154_lp_timer_stop(void)
+void nrf_802154_platform_sl_lptimer_critical_section_enter(void)
 {
 	nrf_802154_sl_mcu_critical_state_t state;
 
 	nrf_802154_sl_mcu_critical_enter(state);
 
-	m_is_running = false;
-	(void)z_nrf_rtc_timer_compare_int_lock(m_rtc_channel);
+	m_in_critical_section++;
+	__DMB();
+
+	if (m_in_critical_section == 1UL) {
+		(void)z_nrf_rtc_timer_compare_int_lock(m_rtc_channel);
+		(void)z_nrf_rtc_timer_overflow_int_lock();
+	}
 
 	nrf_802154_sl_mcu_critical_exit(state);
 }
 
-void nrf_802154_lp_timer_sync_start_now(void)
+void nrf_802154_platform_sl_lptimer_critical_section_exit(void)
 {
-	/* Currently not supported */
-}
+	nrf_802154_sl_mcu_critical_state_t state;
 
-void nrf_802154_lp_timer_sync_start_at(uint32_t t0, uint32_t dt)
-{
-	(void)t0;
-	(void)dt;
-	/* Currently not supported */
-}
+	nrf_802154_sl_mcu_critical_enter(state);
 
-void nrf_802154_lp_timer_sync_stop(void)
-{
-	/* Currently not supported */
-}
+	/*! TODO: Shall this be an assertion?
+	 *  @todo Shall this be an assertion?
+	 */
+	assert(m_in_critical_section > 0UL);
 
-uint32_t nrf_802154_lp_timer_sync_event_get(void)
-{
-	/* Currently not supported */
-	return 0;
-}
+	z_nrf_rtc_timer_compare_int_unlock(m_rtc_channel,
+					   m_is_running && (m_in_critical_section == 1UL));
+	z_nrf_rtc_timer_overflow_int_unlock(m_in_critical_section == 1UL);
```

Comment:
Is `z_nrf_rtc_timer_overflow_int_unlock` implemented anywhere? How does it work?